notebooks/clean-retraining/SAMO_Stage2.ipynb

```
@@ -0,0 +1,645 @@
+        "            # Create multi-label for complex emotions\n",
+        "            labels = [0.0] * NUM_EMOTIONS\n",
+        "            labels[emotion_to_idx[emotion]] = 1.0\n",
+        "            \n",
+        "            # Add secondary emotions for complexity\n",
+        "            if emotion == 'bittersweet':\n",
+        "                labels[emotion_to_idx['joy']] = 0.7\n",
+        "                labels[emotion_to_idx['sadness']] = 0.6\n",
+        "            elif emotion == 'nostalgia':\n",
+        "                labels[emotion_to_idx['love']] = 0.5\n",
+        "                labels[emotion_to_idx['sadness']] = 0.4\n",
+        "            elif emotion == 'euphoria':\n",
```

Ruff (0.12.2)

First line should end with a period, question mark, or exclamation point

Add closing punctuation

(D415)